packages/components/src/button/toolbar-button.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * External dependencies
+ */
+import { useToolbarItem } from 'reakit/Toolbar';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useContext } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ToolbarContext from '../toolbar-context';
+
+function ToolbarButton( { as: Component = 'button', ...props }, ref ) {
+	const context = useContext( ToolbarContext );
+
+	// https://reakit.io/docs/composition/#props-hooks
+	const itemHTMLProps = useToolbarItem( context, {
+		ref,
+		...props,
+		// With this attribute, can check if `ToolbarButton` is used within the
+		// tree and then decide whether to use the accessible Toolbar (which only
+		// accepts `ToolbarButton` as toolbar items) or fallback to the legacy
+		// `NavigableToolbar`, which accepts any tabbable element.
+		'data-toolbar-button': true,
```

Comment:
It would be better if we could mark legacy focusable elements with the data attribute but I guess this is way much easier. I also can't think myself of any better way of doing it.

Reply:
The only way I can think is using [`MutationObserver`](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver). That's also what I would try to use to make `NavigableContainer` add `tabindex="-1"` to inactive items.